ivy.el

```diff
@@ -1927,8 +1930,14 @@ This is useful for recursive `ivy-read'."
          (require-match (ivy-state-require-match state))
          (caller (or (ivy-state-caller state) this-command))
          (sort (or (ivy-state-sort state) (assoc caller ivy-sort-functions-alist)))
-         (initial-input (or (ivy-state-initial-input state)
-                            (cdr (assq caller ivy-initial-inputs-alist))))
+         (initial-input
+          (progn
+            (or (ivy-state-initial-input state)
+                (let ((init (cdr (assq caller ivy-initial-inputs-alist))))
+                  (cond ((functionp init)
+                         (funcall init))
+                        (t
+                         init))))))
```

Comment:
> Should the `funcall` be wrapped in `ignore-errors`, or is it up to the user to ensure their customisations are sensible?

Probably best to let any errors propagate, so that the user is made aware of the problem. Catching errors and warning the user would be overkill, IMO.